File: Practice/Code Golf/Power of Thor.py
# 154
X,Y,x,y=[int(i)for i in input().split()]
while 1:m,y=(((["",y],["N",y-1])[y>Y]),["S",y+1])[y<Y];m,x=((([m,x],[m+"W",x-1])[x>X]),[m+"E",x+1])[x<X];print(m)

File: Compete/Code Golf/Power of Thor/Power of Thor.js
// 1 - 100
[X,Y,x,y]=readline().split(' ');
for(;;){
    print((Y>y++?"S":"")+(X>x++?"E":X<--x?(x--,"W"):""))
}

/* 94
[X,Y,x,y]=readline().split(' ');
for(;;){print((Y>y++?"S":"")+(X>x++?"E":X<--x?(x--,"W"):""))}
*/

// 1.1 - 87
for([X,Y,x,y]=readline().split` `;;)
    print((y++<Y?'S':'')+(x--<X?'E':x++>X?'W':''))

/* 82
for([X,Y,x,y]=readline().split` `;;)print((y++<Y?'S':'')+(x--<X?'E':x++>X?'W':''))
*/